File: ICARUS/Models/Icarus/Form/Input.cs
using ICARUS.Models.Icarus;
using ICARUS.Models.Icarus.Elements;
using ICARUS.Models.Icarus.Elements.Attributes;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.RegularExpressions;
using System.Web.Mvc;

namespace ICARUS.Models {

    /// <summary>
    /// A Form Element (INPUT, SELECT etc)
    /// </summary>
    public class Input : FormElement {

        /// <summary>
        /// Generic FormElement Constructor for Entity
        /// </summary>
        public Input() : base() {
            var guid = "INPUT_" + Guid.NewGuid();
            this.element = "INPUT";
            this.hasTab = 0;
            this.showHeader = 0; // temp
            this.label = guid;
            this.attributes.Add("name", guid);
            this.className = "INPUT";
        }

        /// <summary>
        /// Constructs an FormElement with the given id and label attributes already set
        /// </summary>
        /// <param name="formPost"></param>
        public Input(FormPost formPost) : base(formPost) {

        }
    }
}

File: Scripts/src/icarus/model/EL/input/INPUT.js
/** @module */
import EL, { ATTRIBUTES, MODEL } from '../EL.js';
import { ATTR, MODELS } from '../../../enums/DATAELEMENTS.js';
/** An INPUT element */
export default class INPUT extends EL {
    /** Constructs a generic INPUT Element
        @param {EL} node Node
        @param {InputModel} model Model
    */
    constructor(node, model = MODELS.input()) {
        super(node, 'INPUT', model);
	}
}
export { ATTR, ATTRIBUTES, EL, MODEL, MODELS }